core/modules/rproc/ProtoRowBuffer.h

```diff
@@ -23,14 +23,121 @@
 #ifndef LSST_QSERV_RPROC_PROTOROWBUFFER_H
 #define LSST_QSERV_RPROC_PROTOROWBUFFER_H
 
+// Class header
+#include "rproc/ProtoRowBuffer.h"
+
+// System headers
+#include <limits>
+
+
+// Qserv headers
 #include "mysql/RowBuffer.h"
 #include "proto/worker.pb.h"
+#include "sql/Schema.h"
 
 namespace lsst {
 namespace qserv {
 namespace rproc {
-/// Construct a RowBuffer with a Result message row source.
-mysql::RowBuffer::Ptr newProtoRowBuffer(proto::Result& r);
+
+
+/// ProtoRowBuffer is an implementation of RowBuffer designed to allow a
+/// LocalInfile object to use a Protobufs Result message as a row source
+class ProtoRowBuffer : public mysql::RowBuffer {
+public:
+    ProtoRowBuffer(proto::Result& res, int jobId, std::string const& jobIdColName,
+                   std::string const& jobIdSqlType, int jobIdMysqlType);
+    virtual unsigned fetch(char* buffer, unsigned bufLen);
+    std::string dump() override;
+
+    /// Escape a bytestring for LOAD DATA INFILE, as specified by MySQL doc:
+    /// https://dev.mysql.com/doc/refman/5.1/en/load-data.html
+    /// This is limited to:
+    /// Character    Escape Sequence
+    /// \0     An ASCII NUL (0x00) character
+    /// \b     A backspace character
+    /// \n     A newline (linefeed) character
+    /// \r     A carriage return character
+    /// \t     A tab character.
+    /// \Z     ASCII 26 (Control+Z)
+    /// \N     NULL
+    ///
+    /// @return the number of bytes written to dest
+    template <typename Iter, typename CIter>
+    static inline int escapeString(Iter destBegin, CIter srcBegin, CIter srcEnd) {
+        //mysql_real_escape_string(_mysql, cursor, col, r.lengths[i]);
+        // empty string isn't escaped
+        if (srcEnd == srcBegin) return 0;
+        assert(srcEnd - srcBegin > 0);
+        assert(srcEnd - srcBegin < std::numeric_limits<int>::max() / 2);
+        Iter destI = destBegin;
+        for(CIter i = srcBegin; i != srcEnd; ++i) {
+            switch(*i) {
+              case '\0':   *destI++ = '\\'; *destI++ = '0'; break;
+              case '\b':   *destI++ = '\\'; *destI++ = 'b'; break;
+              case '\n':   *destI++ = '\\'; *destI++ = 'n'; break;
+              case '\r':   *destI++ = '\\'; *destI++ = 'r'; break;
+              case '\t':   *destI++ = '\\'; *destI++ = 't'; break;
+              case '\032': *destI++ = '\\'; *destI++ = 'Z'; break;
+              default: *destI++ = *i; break;
```

Comment:
I moved this class from the cc file and don't what to mess too much with something I believe to be working. I agree that it would be good to know why those cases are omitted.